stdlib/2/os/path.pyi

```diff
@@ -22,25 +28,32 @@ devnull = ...  # type: str
 def abspath(path: AnyStr) -> AnyStr: ...
 def basename(path: AnyStr) -> AnyStr: ...
 
-def commonprefix(list: List[AnyStr]) -> AnyStr: ...
+if sys.version_info >= (3, 5):
+    def commonpath(paths: Sequence[AnyStr]) -> AnyStr: ...
+
+# NOTE: Empty lists results in '' (str) regardless of contained type.
+# Also, in Python 2 mixed sequences of Text and bytes results in either Text or bytes
```

Comment:
Oh right, I forgot about that promotion. Seems like this is the best we can do then. Thanks for explaining!